rllib/algorithms/ppo/utils/env_runner.py

```diff
@@ -0,0 +1,471 @@
+import gymnasium as gym
+import numpy as np
+import tree
+
+from functools import partial
+from typing import Dict, List, Optional, Tuple
+from collections import defaultdict
+from ray.experimental.tqdm_ray import tqdm
+from ray.rllib.algorithms.algorithm_config import AlgorithmConfig
+from ray.rllib.core.models.base import STATE_IN, STATE_OUT
+from ray.rllib.core.rl_module.marl_module import MultiAgentRLModule
+from ray.rllib.evaluation.metrics import RolloutMetrics
+from ray.rllib.env.env_runner import EnvRunner
+from ray.rllib.env.utils import _gym_env_creator
+from ray.rllib.policy.sample_batch import DEFAULT_POLICY_ID, SampleBatch
+from ray.rllib.utils.annotations import override
+from ray.rllib.utils.framework import try_import_tf
+from ray.rllib.utils.replay_buffers.episode_replay_buffer import _Episode as Episode
+from ray.rllib.utils.rl_module import foreach_module
```

Comment:
I didn't know we had this utility, but wouldn't it be more intuitive to have the same logic as a method inside of the `MARLModule` class?

Reply:
Analogous to `WorkerSet.foreach_worker`.

Reply:
I totally agree, but I first walked the sideway to keep generic code as clean as possible. I will move this over.